doc/rapidcheck.md

```diff
@@ -0,0 +1,114 @@
+# RapidCheck property-based testing for Bitcoin Core
+
+## Concept
+
+Property-based testing is experimentally being added to Bitcoin Core with
+[RapidCheck](https://github.com/emil-e/rapidcheck), a C++ framework for
+property-based testing inspired by the Haskell library
+[QuickCheck](https://hackage.haskell.org/package/QuickCheck).
+
+RapidCheck performs random testing of program properties. A specification of the
+program is given in the form of properties which functions should satisfy, and
+RapidCheck tests that the properties hold in a large number of randomly
+generated cases.
+
+If an exception is found, RapidCheck tries to find the smallest case, for some
+definition of smallest, for which the property is still false and displays it as
+a counter-example. For example, if the input is an integer, RapidCheck tries to
+find the smallest integer for which the property is false.
+
+## Running
+
+Once RapidCheck is installed correctly, Bitcoin Core will automatically run the
+property-based tests with the unit tests during `make check`, unless the
+`--without-rapidcheck` flag is passed when configuring.
+
+For more information, run `./configure --help` and see `--with-rapidcheck` under
+"Optional Packages".
+
+## Setup
+
+The following instructions have been tested with Linux Debian 4.19/GCC 8.3 and
+macOS 10.14.6/AppleClang 10.
+
+1. Ensure cmake is installed with `apt-get install cmake` on Linux or `brew
+   install cmake` on macOS.
+
+2. Clone the RapidCheck source code and cd into the repository.
+
+    ```shell
+    git clone https://github.com/emil-e/rapidcheck.git
+    cd rapidcheck
+    ```
+
+3. Build RapidCheck.
+
+    ```shell
+    cmake -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=true -DRC_ENABLE_BOOST_TEST=ON $(pwd)
+    make && make install
+    ```
+
+    You should see:
+
+    ```shell
+    -- The CXX compiler identification is <GNU/AppleClang/etc x.y.z>
```

Comment:
I don't think we need all of this build output here.

Reply:
I think you're right. The build output was helpful for me in debugging my CMake command but it is resolved now.